### PR TITLE
Added Ultisnips builtin completion source

### PR DESCRIPTION
**Pull request by RishabhRD**
Hello @hrsh7th , as discussed in #65 , This PR adds ultisnips completion source. If something is missing from the standard, I would modify it accordingly. :sweat_smile:

### CODE REVIEW

File: after/plugin/compe_ultisnips.vim [hunk 1]
```diff
@@ -0,0 +1,9 @@
+if exists('g:loaded_compe_ultisnips')
+  finish
+endif
+let g:loaded_compe_ultisnips = v:true
+
+if exists('g:loaded_compe')
```
Comment (hrsh7th) [inline after/plugin/compe_ultisnips.vim:6]: Could you check ultisnips installed or not?

File: lua/compe_ultisnips/init.lua [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+function M:datermine(context)
+  return compe.helper.datermine(context)
+end
+
+function M:complete(args)
+  if vim.fn.exists("*UltiSnips#SnippetsInCurrentScope") == 0 then
+    args.abort()
+    return
+  end
```
Comment (hrsh7th) [inline lua/compe_ultisnips/init.lua:21]: It should be moved to datermine or just removed.

File: lua/compe_ultisnips/init.lua [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  local receivedSnippets = vim.call('UltiSnips#SnippetsInCurrentScope')
```
Comment (hrsh7th) [inline lua/compe_ultisnips/init.lua:22]: Please use snake_case instead.

File: lua/compe_ultisnips/init.lua [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  if vim.tbl_isempty(receivedSnippets) then
+    args.abort()
+    return
+  end
+  local completionList = {}
+  for key, value in pairs(receivedSnippets) do
+    local item = {
+      word =  key,
+      abbr =  key,
+      userdata = value,
```
Comment (hrsh7th) [inline lua/compe_ultisnips/init.lua:32]: it should be `user_data` because vim's completed-item uses it.

File: lua/compe_ultisnips/init.lua [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+end
+
+function M:documentation(args)
+  local completedItem = args.completed_item
+  local userdata = completedItem.userdata
+  if userdata == nil or userdata == '' then
+    args.abort()
+    return
+  end
+  args.callback(userdata)
```
Comment (hrsh7th) [inline lua/compe_ultisnips/init.lua:50]: LGTM.

File: after/plugin/compe_ultisnips.vim (hunk 1 above)
Comment (RishabhRD) [reply after/plugin/compe_ultisnips.vim:6]: Yeah! I would include the check using ``match`` function of vimscript

File: lua/compe_ultisnips/init.lua (hunk 4 above)
Comment (RishabhRD) [reply lua/compe_ultisnips/init.lua]: OK! I didn't know about it. Just changing it. :smiley:

File: lua/compe_ultisnips/init.lua (hunk 3 above)
Comment (RishabhRD) [reply lua/compe_ultisnips/init.lua]: Sure! :sweat_smile:

File: lua/compe_ultisnips/init.lua (hunk 2 above)
Comment (RishabhRD) [reply lua/compe_ultisnips/init.lua:22]: Once checked if ultisnips is installed or not while registering I don't think there would be any use for it. So, I would remove it.

**Comment by RishabhRD** [comment]
@hrsh7th I have added the changes you asked for. Please review it, if something is missing. :smile:

**Comment by tami5** [comment]
Wow @RishabhRD, support for Ultisnips is very important as many vim user depend on. Thanks

**Comment by hrsh7th** [comment]
LGTM. Thank you ~.

**Pull request merged**